admin/ajax/class-recalculate-scores-ajax.php

```diff
@@ -0,0 +1,197 @@
+			array_merge(
+				$this->query_fields,
+				array(
+					'posts_per_page' => $this->posts_per_page,
+					'paged'          => $paged,
+					'posts'          => array(),
+				)
+			)
+		);
+
+		if ( $posts = $post_query->get_posts() ) {
+			$this->options = WPSEO_Options::get_all();
+
+			$parsed_posts = $this->parse_posts( $posts );
+
+			$response  = array(
+				'posts'       => $parsed_posts,
+				'total_posts' => count( $parsed_posts ),
+				'next_page'   => ( $paged + 1 ),
+			);
+
+			return json_encode( $response );
+		}
+
+		return '';
+	}
+
+	/**
+	 * Parsing the posts with the value we need
+	 *
+	 * @param array $posts
+	 *
+	 * @return array
+	 */
+	private function parse_posts( array $posts ) {
+		$parsed_posts = array();
+		foreach ( $posts as $post ) {
+			$parsed_posts[] = $this->post_to_response( $post );
+		}
+
+		return $parsed_posts;
+	}
+
+	/**
+	 * @param WP_Post $post
+	 *
+	 * @return array
+	 */
+	private function post_to_response( WP_Post $post ) {
+		$focus_keyword = WPSEO_Meta::get_value( 'focuskw', $post->ID );
+
+		return array(
+			'post_id'       => $post->ID,
+			'text'          => $post->post_content,
+			'keyword'       => $focus_keyword,
+			'url'           => urldecode( $post->post_name ),
+			'pageTitle'     => apply_filters( 'wpseo_title', wpseo_replace_vars( $this->get_title( $post->ID, $post->post_type ), $post ) ),
+			'meta'          => apply_filters( 'wpseo_metadesc', wpseo_replace_vars( $this->get_meta_description( $post->ID, $post->post_type ), $post ) ),
+			'keyword_usage' => array(
+				$focus_keyword => WPSEO_Meta::keyword_usage( $focus_keyword, $post->ID ),
+			),
+		);
+	}
+
+	/**
+	 * Getting the title for given post
+	 *
+	 * @param integer $post_id
+	 * @param string  $post_type
+	 *
+	 * @return mixed|string
+	 */
+	private function get_title( $post_id, $post_type ) {
+		if ( ( $title = WPSEO_Meta::get_value( 'title', $post_id )  ) !== '' ) {
+			return $title;
+		}
+
+		if ( $default_from_options = $this->default_from_options( 'title', $post_type ) ) {
+			return str_replace( ' %%page%% ', ' ', $default_from_options );
+		}
+
+		return '%%title%%';
+	}
+
+	/**
+	 * Getting the meta description for given post
+	 *
+	 * @param integer $post_id
+	 * @param string  $post_type
+	 *
+	 * @return bool|string
+	 */
+	private function get_meta_description( $post_id, $post_type ) {
+		if ( ( $meta_description = WPSEO_Meta::get_value( 'metadesc', $post_id ) ) !== '' ) {
+			return $meta_description;
+		}
+
+		if ( $default_from_options = $this->default_from_options( 'metadesc', $post_type ) ) {
+			return $default_from_options;
+		}
+
+		return '';
+	}
+
+	/**
+	 * Getting default from the options for given field
+	 *
+	 * @param string $field
+	 * @param string $post_type
+	 *
+	 * @return bool|string
+	 */
+	private function default_from_options( $field, $post_type ) {
+		if ( isset( $this->options[ $field . '-' . $post_type ] ) && $this->options[ $field . '-' . $post_type ] !== '' ) {
```

Comment:
Assign `$this->options[ $field . '-' . $post_type ]` to a local variable and use that 3 times. Cleaner code.